src/components/authoring/definition-editor.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import * as css from "./definition-editor.scss";
+import { IWordDefinition } from "../types";
+import Button from "./button";
+
+interface IProps {
+  onSave: (definition: IWordDefinition) => void;
+  onCancel: () => void;
+  initialDefinition?: IWordDefinition;
+}
+
+interface IState {
+  definition: IWordDefinition;
+  error: string;
+}
+
+export default class DefinitionEditor extends React.Component<IProps, IState> {
+  public state: IState = {
+    definition: this.props.initialDefinition || {
+      word: "",
+      definition: "",
+      image: "",
+      imageCaption: "",
+      video: "",
+      videoCaption: ""
+    },
+    error: ""
+  };
+
+  public render() {
+    const { onCancel, initialDefinition } = this.props;
+    const { definition, error } = this.state;
+    return (
+      <div className={css.editor}>
+        <table>
+          <tbody>
+            <tr>
+              <td>{initialDefinition ? "Edit word" : "Word"}</td>
+              <td>
+                {
+                  initialDefinition ?
+                    initialDefinition.word :
```

Comment:
This is working fine in the demo, so it works, but I am confused.

If this is the intialDefinition, display the word value (not in an editable input area), otherwise show a text box with the word as the value.  Is that right?

Reply:
Yes. When the `initialDefinition` property is present, it means that we are editing an existing definition. If it's not, then we're adding a new one. Perhaps I should add a comment explaining that.

The reason that you cannot edit "word" property of an existing definition is to avoid conflicts with another one. If you changed "word" to one which is already there and click "Save", the behavior wouldn't make much sense. We could just have two entries with the same "word", but I'm making sure that you cannot author it to avoid possible confusion (e.g. author editing the first entry while application respects the last one, or the other way).

Reply:
Right -- sorry in hindsight this makes more sense.  Thank you for the explainer.